library/lib_orbital_maneuvers.ks

```diff
@@ -0,0 +1,260 @@
+	// n is mean angular velocity
+	local t1 is m1/n.
+	local t2 is m2/n.
+	// t1 and t2 are times with regard to some, non-disclosed epoch,
+	// at which orbitable will be at true anomaly a1 or a2 respectively
+	local diff is t2-t1.
+	if diff<0{ // ETA must be positive, so switch to next orbit
+		set diff to orb:period+diff.
+	}
+	else if diff>orb:period{ // we can do one full orbit less
+		set diff to diff-orb:period.
+	}
+	return diff.
+}
+
+// This function takes two orbits CONTAINED IN THE SAME PLANE and returns
+// a list of at most three elements:
+// - first one is a "status" - if it's equal to -1, then obt1 is fully
+// contained within obt2, vice versa for +1. Otherwise, it is equal to
+// 0 and the list contains two more elements
+// - two other elements, existing only if orbits cross at some points,
+// represent the time remaining until ship orbiting obt1 will pass common
+// orbit point. This should be the point at which you would do the burn
+// to randez-vous
+function get_orbit_intersections{
+	parameter
+		obt1,
+		obt2.
+
+	local e1 is obt1:eccentricity.
+	local e2 is obt2:eccentricity.
+	local a1 is obt1:semimajoraxis.
+	local a2 is obt2:semimajoraxis.
+	local B is a1/a2*(1-e1*e1)/(1-e2*e2).
+	local delta is obt2:argumentofperiapsis-obt1:argumentofperiapsis.
+	local gamma is arctan2(e1-B*e2*cos(delta), -B*e2*sin(delta)).
+	local S is (1-B)*cos(gamma)/B/e2/sin(delta).
+	if S > 1-0.00001{
+		return list(1).
+	}
+	else if S < -1+0.00001{
+		return list(-1).
+	}
+	local beta1 is arcsin(S)-gamma.
+	local beta2 is 180-arcsin(S)-gamma.
+	return list(
+		0, 
+		time_to_true_anomaly(obt1,beta1),
+		time_to_true_anomaly(obt1,beta2)
+	).
+}
+
+// This function will put a randez-vous node at orbit intersection 
+// between your orbit and tgt's orbit, if it exists. It assumes orbits
+// share common plane.
+function match_orbits{
+	parameter tgt.
+	
+	local l is get_orbit_intersections(obt,tgt:obt).
+	if l[0]<>0{
+		return 0.
+	}
+	local t is l[1].
+	if l[2]<t{
+		set t to l[2].
+	}
+
+	local actual_vel is velocityat(ship,time:seconds+t):orbit.
+	local intersection_r is body:position-positionat(ship,time:seconds+t).
+	
+	local tgt_r is body:position-tgt:position.
+	local tgt_normal is VCRS(tgt:velocity:orbit,tgt_r).
+	local side is VCRS(tgt_normal,tgt_r).
+	local true_anomaly is VANG(intersection_r,tgt_r).
+	if VDOT(side,intersection_r)<0{
+		set true_anomaly to -true_anomaly.
+	}
+	set true_anomaly to true_anomaly+tgt:obt:trueanomaly.
+	local t2 is time_to_true_anomaly(tgt:obt,true_anomaly).
+
+	local expected_vel is velocityat(tgt,time:seconds+t2):orbit.
+	return make_node_t_deltav(time:seconds+t,expected_vel-actual_vel).
+}
+
+// This function takes two identical orbits as arguments (though they
+// differ in ship's current position). Function returns a node, that in
+// N orbits (N is integer parameter) will match phase of both ships (put
+// them in almost exact same place).
+function match_orbital_phase{
+	parameter
+		orb1,
```

Comment:
But really, it is unlikely this function is useful in any case other than my ship as first orbit - so refactoring this function to other parameters.